ospf6d/ospf6_auth_trailer.c

```diff
@@ -0,0 +1,942 @@
+       "debug authentication rx\n")
+{
+	int auth_opt_idx = 3;
+
+	if (argc == 5) {
+		if (!strncmp(argv[auth_opt_idx]->arg, "t", 1))
+			OSPF6_DEBUG_AUTH_TX_OFF();
+		else if (!strncmp(argv[auth_opt_idx]->arg, "r", 1))
+			OSPF6_DEBUG_AUTH_RX_OFF();
+	} else {
+		OSPF6_DEBUG_AUTH_TX_OFF();
+		OSPF6_DEBUG_AUTH_RX_OFF();
+	}
+
+	return CMD_SUCCESS;
+}
+
+int config_write_ospf6_debug_auth(struct vty *vty)
+{
+	if (IS_OSPF6_DEBUG_AUTH_TX)
+		vty_out(vty, "debug ospf6 authentication tx\n");
+	if (IS_OSPF6_DEBUG_AUTH_RX)
+		vty_out(vty, "debug ospf6 authentication rx\n");
+	return 0;
+}
+
+void install_element_ospf6_debug_auth(void)
+{
+	install_element(ENABLE_NODE, &debug_ospf6_auth_cmd);
+	install_element(ENABLE_NODE, &no_debug_ospf6_auth_cmd);
+	install_element(CONFIG_NODE, &debug_ospf6_auth_cmd);
+	install_element(CONFIG_NODE, &no_debug_ospf6_auth_cmd);
+}
+
+/* Clear the specified interface structure */
+static void ospf6_intf_auth_clear(struct vty *vty, struct interface *ifp)
+{
+	struct ospf6_interface *oi;
+
+	if (!if_is_operative(ifp))
+		return;
+
+	if (ifp->info == NULL)
+		return;
+
+	oi = (struct ospf6_interface *)ifp->info;
+
+	if (IS_OSPF6_DEBUG_INTERFACE)
+		zlog_debug(
+			"Interface %s: clear authentication rx/tx drop counters",
+			ifp->name);
+
+	/* Reset the interface rx/tx drop counters */
+	oi->at_data.tx_drop = 0;
+	oi->at_data.rx_drop = 0;
+}
+
+/* Clear interface */
+DEFUN(clear_ipv6_ospf6_intf_auth,
+      clear_ipv6_ospf6_intf_auth_cmd,
+      "clear ipv6 ospf6 auth-counters interface [IFNAME]",
+      CLEAR_STR
+      IP6_STR
+      OSPF6_STR
+      "authentication rx/tx drop counters\n"
+      INTERFACE_STR
+      IFNAME_STR)
+{
+	struct vrf *vrf = vrf_lookup_by_id(VRF_DEFAULT);
+	int idx_ifname = 5;
+	struct interface *ifp;
+
+	if (argc == 5) { /* Clear all the ospfv3 interfaces. */
+		FOR_ALL_INTERFACES (vrf, ifp)
+			ospf6_intf_auth_clear(vty, ifp);
+	} else { /* Interface name is specified. */
+		ifp = if_lookup_by_name(argv[idx_ifname]->arg, VRF_DEFAULT);
+		if (ifp  == NULL) {
+			vty_out(vty, "Error: No such Interface: %s\n",
+				argv[idx_ifname]->arg);
+			return CMD_WARNING;
+		}
+		ospf6_intf_auth_clear(vty, ifp);
+	}
+
+	return CMD_SUCCESS;
+}
+
+void install_element_ospf6_clear_intf_auth(void)
+{
+	install_element(ENABLE_NODE, &clear_ipv6_ospf6_intf_auth_cmd);
+}
+
+enum ospf6_auth_err ospf6_auth_file_storage_exist(void)
+{
+	struct stat buffer;
+	int exist;
+
+	exist = stat(OSPF6_AUTH_SEQ_NUM_FILE, &buffer);
+	if (exist == 0)
+		return OSPF6_AUTH_FILE_EXIST;
+	else
+		return OSPF6_AUTH_FILE_DO_NOT_EXIST;
+}
+
+bool ospf6_auth_file_storage_rec_add(uint32_t vrf_id, uint32_t seqnum_h)
+{
+	FILE *fp;
+	int ret = 0;
+	struct ospf6_at_seq_rec_t rec;
+
+	fp = fopen(OSPF6_AUTH_SEQ_NUM_FILE, "ab");
+	if (!fp) {
+		zlog_err("Failed to open file %s ab mode",
+			 OSPF6_AUTH_SEQ_NUM_FILE);
+		return false;
+	}
+
+	rec.vrf_id = vrf_id;
+	rec.seqnum_h = seqnum_h;
+	ret = fwrite(&rec, sizeof(rec), 1, fp);
```

Comment:
No unstructured binary dumps please.

Make it JSON like `ospf_gr_nvm_*` functions.

Reply:
Since it's a security feature, Don't you think the information should be in non readable format ?